test/unit/module/core/first_true.cpp

```diff
@@ -16,111 +16,90 @@ TTS_CASE_TPL("Check eve::first_true return type", eve::test::simd::all_types)
+  TTS_EQUAL(eve::first_true[eve::ignore_none](v), manual_first_true(v, true));
 
-  // ignore last
+  if constexpr (eve::simd_value<T>)
   {
-    l_t x(false);
-
-    for( int i = 0; i != T::size(); ++i )
-    {
-      x.set(i, true);
-
-      auto ilast        = eve::ignore_last(T::size() - i - 1);
-      auto ilast_bigger = eve::ignore_last(T::size() - i);
-
-      TTS_EQUAL(eve::first_true[ilast](x), i);
-      TTS_EQUAL(eve::first_true[ilast_bigger](x), std::nullopt);
-
-      if( i > 0 )
-      {
-        auto iextrema = ilast && eve::ignore_first(i - 1);
-        TTS_EQUAL(eve::first_true[iextrema](x), i);
-
-        iextrema = ilast_bigger && eve::ignore_first(i - 1);
-        TTS_EQUAL(eve::first_true[iextrema](x), std::nullopt);
-      }
+    T m = [](auto i, auto) { return i % 2 == 0; };
+    TTS_EQUAL(eve::first_true[m](v), manual_first_true(v, m));
+
+    TTS_EQUAL(eve::first_true[eve::ignore_all](v), std::nullopt);
+    TTS_EQUAL(eve::first_true[eve::keep_first(1)](v), manual_first_true(v, eve::keep_first(1)));
+    TTS_EQUAL(eve::first_true[eve::ignore_first(1)](v), manual_first_true(v, eve::ignore_first(1)));
+    TTS_EQUAL(eve::first_true[eve::keep_last(1)](v), manual_first_true(v, eve::keep_last(1)));
+    TTS_EQUAL(eve::first_true[eve::ignore_last(1)](v), manual_first_true(v, eve::ignore_last(1)));
+  }
+}
 
-      x.set(i, false);
-    }
+TTS_CASE("Check eve::first_true on booleans")
+{
+  test_first_true(true);
+  test_first_true(false);
+};
 
-    x = true;
-    TTS_EQUAL(eve::first_true[eve::ignore_last(T::size())](x), std::nullopt);
-  }
+TTS_CASE_TPL("Check eve::first_true behavior on scalars", eve::test::scalar::all_types)
+<typename T>(tts::type<T>)
+{
+  test_first_true(eve::logical<T>{true});
+  test_first_true(eve::logical<T>{false});
 };
 
-TTS_CASE_TPL("Check first_true(top_bits)", eve::test::simd::all_types)
+TTS_CASE_TPL("Check eve::first_true behavior on wides and top_bits", eve::test::simd::all_types)
 <typename T>(tts::type<T>)
 {
-  using logical = eve::logical<T>;
-  TTS_EQUAL(0, eve::first_true(eve::top_bits(logical(true))));
+  constexpr auto cardinal = eve::cardinal_v<T>;
 
-  for( int i = 0; i != T::size() - 1; ++i )
+  for( std::ptrdiff_t j = 0; j < cardinal; ++j )
   {
-    logical       v([=](auto e, auto) { return e > i; });
-    std::optional res = eve::first_true(eve::top_bits(v));
-    TTS_EXPECT(res);
+    eve::logical<T> rhs1 = {}, rhs2 = {}, rhs3 = {};
 
-    int expected = i + 1;
-    TTS_EQUAL(expected, *res);
-  }
+    for( std::ptrdiff_t i = 0; i < cardinal; ++i )
+    {
+      rhs1.set(i, i >= j ? true : false);
+      rhs2.set(i, i <= j ? false : true);
+      rhs3.set(i, i == j ? true : false);
+    }
 
-  TTS_EXPECT_NOT(eve::first_true(eve::top_bits(logical(false))));
+    test_first_true(rhs1);
+    test_first_true(rhs2);
+    test_first_true(rhs3);
+
+    test_first_true(eve::top_bits{rhs1});
+    test_first_true(eve::top_bits{rhs2});
+    test_first_true(eve::top_bits{rhs3});
+  }
```

Comment:
I don't know why did you throw away all of my tests.
My tests were good. 

Where are `ignore_first`?  `ignore_last`? for i bigger than 1? How about testing `ignore_first(n)`